src/pages/countries.js

```diff
@@ -0,0 +1,82 @@
+import { useQuery, gql } from "@apollo/client";
+import { Link, useLocation, useHistory } from "react-router-dom";
+import CountriesList from "../components/CountriesList";
+import { useState, useEffect } from "react";
+
+export const COUNTRIES_QUERY = gql`
+  query allCountries($limit: Int, $skip: Int) {
+    countries(limit: $limit, skip: $skip) {
+      id
+      name
+      vatRate
+      languages {
+        id
+        name
+      }
+    }
+  }
+`;
+
+function useQueryUrl() {
+  return new URLSearchParams(useLocation().search);
+}
+
+const Countries = (props) => {
+  const query = useQueryUrl();
+  const history = useHistory();
+
+  const countriesPerPage = 5;
+  const [page, setPage] = useState(query.get("page"));
+
+  useEffect(() => {
+    setPage(query.get("page"));
+  }, [query.get("page")]);
+
+  const variables = {
+    variables: { limit: countriesPerPage, skip: page * countriesPerPage },
+  };
+
+  const { loading, error, data } = useQuery(COUNTRIES_QUERY, variables);
+
+  const getNextPageNumber = () => {
+    const currPage = parseInt(query.get("page")) || 0;
```

Comment:
i think that this variable can be. extracted with `useMemo(() => {})` hook

Reply:
I changed the currPage into a variable extracted from useMemo hook in incoming commit